pgmpy/estimators/TreeSearch.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+import networkx as nx
+
+from pgmpy.estimators import StructureEstimator
+from sklearn.metrics import mutual_info_score
+from pgmpy.base import DAG
+
+
+class TreeSearch(StructureEstimator):
```

Comment:
I think TreeSearch and NaiveBayesTreeSearch can be combined together by adding some extra parameters to the `estimate` method maybe: `return_type`. It will remove boilerplate code and also make my life easier to maintain. :D

Reply:
I agree and I think it may call for more.  Now that I’ve implemented TAN and the Chow-Liu algorithm, I’m tempted to roll naive bayes into the mix as well.  (I know, I did say before that naive bayes is not a special case of TAN but both are special cases of FAN or ETAN.)  However, in terms of coding, they are like Lego blocks and each block can be a standalone component:

- TAN = Chow-Liu + Naive Bayes

So, my questions are:

1. Do you see Naive Bayes as belonging to “estimators” like TAN and Chow-Liu even though it doesn’t do much structure learning from data.
2. What would be the name of this generic class that can do TAN, Chow-Liu (and Naive Bayes and maybe FAN or ETAN in the future)?

Reply:
We already have a model class for Naive Bayes (https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/models/NaiveBayes.py). So, I would suggest modifying it to be able to represent FAN models with special cases for Naive Bayes and TAN. The estimators should stay in the `estimators` dir and be completely separate from the model class, and should just return an instance of the learned model.